tools/plotjuggler/juggle.py

```diff
@@ -4,20 +4,29 @@
 import multiprocessing
 import subprocess
 import argparse
+import re
 from tempfile import NamedTemporaryFile
 
 from common.basedir import BASEDIR
 from selfdrive.test.process_replay.compare_logs import save_log
 from tools.lib.api import CommaApi
 from tools.lib.auth_config import get_token
 from tools.lib.robust_logreader import RobustLogReader
-from tools.lib.route import Route
+from tools.lib.route import Route, RouteSegmentName, SEGMENT_NAME_RE, ROUTE_NAME_RE
 from urllib.parse import urlparse, parse_qs
 
 juggle_dir = os.path.dirname(os.path.realpath(__file__))
 
 DEMO_ROUTE = "4cf7a6ad03080c90|2021-09-29--13-46-36"
 
+def get_route_and_segment_num(name):
```

Comment:
seems like this should be handled in RouteSegmentName, then pass in a RouteSegmentName instead of a string to juggle_route?

Reply:
yeah I can add a parser that parses route and segment names